contracts/staking/ValidatorMessages.sol

```diff
@@ -367,6 +367,105 @@ library ValidatorMessages {
         return (validationID, nonce, weight);
     }
 
+    /**
+     * @notice Packs a SubnetValidatorWeightUpdateMessage into a byte array.
+     * The message format specification is:
+     * +--------------+----------+----------+
+     * |      codecID :   uint16 |  2 bytes |
+     * +--------------+----------+----------+
+     * |       typeID :   uint32 |  4 bytes |
+     * +--------------+----------+----------+
+     * | validationID : [32]byte | 32 bytes |
+     * +--------------+----------+----------+
+     * |        nonce :   uint64 |  8 bytes |
+     * +--------------+----------+----------+
+     * |       weight :   uint64 |  8 bytes |
+     * +--------------+----------+----------+
+     *                           | 54 bytes |
+     *                           +----------+
+     */
+    function packSubnetValidatorWeightUpdateMessage(
+        bytes32 validationID,
+        uint64 nonce,
+        uint64 weight
+    ) internal pure returns (bytes memory) {
+        bytes memory res = new bytes(54);
+        // Pack the codec ID.
+        for (uint256 i; i < 2; ++i) {
+            res[i] = bytes1(uint8(CODEC_ID >> (8 * (1 - i))));
+        }
+        // Pack the type ID.
+        for (uint256 i; i < 4; ++i) {
+            res[i + 2] =
+                bytes1(uint8(SUBNET_VALIDATOR_WEIGHT_UPDATE_MESSAGE_TYPE_ID >> (8 * (3 - i))));
+        }
+        // Pack the validation ID.
+        for (uint256 i; i < 32; ++i) {
+            res[i + 6] = bytes1(uint8(uint256(validationID >> (8 * (31 - i)))));
+        }
+        // Pack the nonce.
+        for (uint256 i; i < 8; ++i) {
+            res[i + 38] = bytes1(uint8(nonce >> (8 * (7 - i))));
+        }
+        // Pack the weight.
+        for (uint256 i; i < 8; ++i) {
+            res[i + 46] = bytes1(uint8(weight >> (8 * (7 - i))));
```

Comment:
I understand that we have always used this way of packing a bytes array. But it still looks error prone to me compared to an approach that doesn't require hard coding magic number indices for each field.

Reply:
Totally agreed. This very likely won't be the byte packing approach we go with, as it's more complex and less gas efficient than using, say ,`abi.encodePacked`. I went with this approach here for consistency, and plan to update all the packing utilities in one shot in a future change.